lib/dry/rails/engine.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Dry
+  module Rails
+    module Engine
+      # Set container block that will be evaluated in the context of the container
+      #
+      # @param name [Symbol]
+      # @return [self]
+      #
+      # @api public
+      def self.container(name, &block)
+        _container_blocks[name] << block
```

Comment:
this is a bit puzzling tbh, when I do: `Dry::Rails::Engine.container(...) { do something with container }` it's not available until `reload!` call `Finalizer` and evaluates the config again 🤔 (_"worked like that before"_)

would the same happen if container is used eg. in `before_filter` - changes done to container would _"leak"_ into next request? this doesn't sound right... 😅

Reply:
@zlw setting up container should only be available during app's booting phase. Once it's done, we should literally freeze its config (which I believe already happens?)